cmd/collector/main.go

```diff
@@ -48,10 +51,30 @@ const (
 )
 
 func main() {
+	serverState := http.StatusServiceUnavailable
+
+	logger, _ := zap.NewProduction()
+	var serverChannel = make(chan os.Signal, 0)
+	signal.Notify(serverChannel, os.Interrupt, syscall.SIGTERM)
+
+	go func() {
+		// Start the health check server, used for reporting the liveness and readiness state for this component
+		logger.Info("Health Check server started", zap.Int("http-port", *builder.HealthCheckHTTPPort))
+		healthCheckMux := http.NewServeMux()
+		healthCheckMux.HandleFunc("/", func(w http.ResponseWriter, _ *http.Request) {
```

Comment:
The advantage of having its own mux is that it won't influence the main mux in any way (metrics, concurrency, ...). Also, if the main mux is stuck/hung for some reason, the healthcheck mux is still able to report the server's status.